### ISSUE TITLE
Setting id in inputOptions removes the default class attribute from generated HTML for default input type

### ISSUE THREAD

**Issue by YasserHassan** [kartik-v/yii2-widget-activeform#34]
Adding `'inputOptions' => ['id' => 'xyz']]` to default text boxes (without '->textInput()') - like in the search forms generated by Gii - removes the `class="form-control"` part from the generated HTML. This does not happen when using Yii's default Bootstrap ActiveField. However, adding '->textInput()' to the code above generates it back.

**Comment by kartik-v** [comment]
Yes the `form-control` class is not ALWAYS APPENDED to allow developers to achieve advanced styling configurations without `form-control` if needed. 

You can try setting class this way - to consider the class in default input options:

```php
$form->field($model, 'attr')->textInput['class' => 'custom-class']);
```

**Comment by kartik-v** [comment]
Just editing my response... 

`\kartik\widgets\ActiveField` has a new `addClass` property that will always be appended to input - to allow developers to achieve advanced styling configurations by just passing space separated classes to this property. The `addClass` property defaults to `form-control`.

**Comment by YasserHassan** [comment]
I tested it and it seems that `addClass` is not applied if '->textInput()' does not exist but `inputOptions' => ['id' => 'xyz']]` exists. So whether you provide a value for `addClass` or leave it as default, it will be ignored in forms like the search form generated by Gii (again if inputOptions is used) until the developer knows how to get around it.

For clarity:
```
// This ignores addClass (even the default) because inputOptions
// exists but textInput() does not.
echo $form->field($model, 'first_name', ['inputOptions' => ['id' => 'first_name']]);

// This is OK because inputOptions does not exist.
echo $form->field($model, 'first_name');

// This is OK with inputOptions because textInput() exists.
echo $form->field($model, 'first_name', ['inputOptions' => ['id' => 'first_name']])->textInput();
```
It's a minor issue but worth noting.

**Comment by kartik-v** [comment]
There is a reason why the `form-control` via `addClass` is not included for `inputOptions` - because it is NOT APPLICABLE for certain active field methods.

Note that this widget has other additional active field inputs unlike yii bootstrap like `staticInput`, `multiselect`, and needs additional treatment for `checkboxButtonGroup`, `radioButtonGroup`, `checkboxList`, `radioList`.

Hence, its individually initialized in options for all relevant methods. The workaround is to have gii generate the textInput implicitly.

**Comment by YasserHassan** [comment]
@kartik-v I think there's a misunderstanding so let me rephrase it. From a web developer's point of view, if a code like `$form->field($model, 'first_name')` generates a proper Bootstrap input with the right class and a default id, why would specifying my own id instead of the default one (this is the only change) make the input lose its CSS class and becomes non-Bootstrap? The result should be the same whether the id is auto generated or provided by the developer. In other words, changing the input id property should not remove the class property.

If this is they way it was meant to be then I'll stop here.

**Comment by kartik-v** [comment]
Ok to summarize - currently you must use it like this instead of input Options if you do not wish to provide a class:

```php
echo $form->field($model, 'first_name')->textInput(['id' => 'first_name']);
```

And here is the explanation. The `form-control` is added to each of the `options` for `textInput`, `input`, `dropdownList`, `listBox` etc..

This class is not added to `inputOptions` - because the styling of other activefield like `staticInput` will be broken.

**Comment by YasserHassan** [comment]
This is a good tip. Thank you.
